siliconcompiler/core.py

```diff
@@ -4050,7 +4011,9 @@ def run(self):
                 indexlist[step] = self.get("option", 'indexlist')
             else:
                 indexlist[step] = self.getkeys('flowgraph', flow, step)
+        return indexlist
 
+    def _resume_steps(self, flow, steplist, indexlist):
```

Comment:
rename this to make it clear, it's doing the preprocessing for the run. 
Also steplist and index list can be gotten from the helper methods and just makes this signature simpler

Reply:
What do you think of `_reset_flow_nodes`?
Same thing as above. Can steplist and indexlist change between function calls?

Reply:
_reset_flow_nodes is fine